### PR TITLE
Feature/query microservices

### PR DESCRIPTION
**Pull request by jwomeara**

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                This allows us to retrieve released microservice artifacts from our github repository.
+            -->
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+            <id>datawave-github-mvn-repo</id>
+            <url>https://raw.githubusercontent.com/NationalSecurityAgency/datawave/mvn-repo</url>
```
Comment (alerman) [inline api/pom.xml]: Use new GHCR Packages

**Review by alerman — APPROVED** [review]

File: api/pom.xml (hunk 1 above)
Comment (jwomeara) [reply api/pom.xml]: good catch!  im on it

**Review by apmoriarty — APPROVED** [review]

File: service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+                QueryException qe = new QueryException(DatawaveErrorCode.MAPREDUCE_JOB_START_ERROR, e);
+                log.error(qe.getMessage(), qe);
+                throw qe;
+            }
+            
+            JobID mapReduceJobId = job.getJobID();
+            log.info("JOB ID: " + mapReduceJobId);
+            
+            // This is used to lookup the query id when the mapreduce callback comes in
+            // TODO: JWO: Should we use a CustomJobEndNotifier that passes the map reduce query id to the update state call instead?
```
Comment (matthpeterson) [inline service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java]: todo

File: service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+            mapReduceQueryStatus.setStackTrace(new String(outputStream.toByteArray(), StandardCharsets.UTF_8));
+        } catch (Exception e) {
+            mapReduceQueryStatus.setErrorCode(DatawaveErrorCode.UNKNOWN_SERVER_ERROR);
+            mapReduceQueryStatus.setFailureMessage(e.getMessage());
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            PrintWriter writer = new PrintWriter(new OutputStreamWriter(outputStream, StandardCharsets.UTF_8));
+            e.printStackTrace(writer);
+            writer.close();
+            mapReduceQueryStatus.setState(MapReduceQueryStatus.MapReduceQueryState.FAILED);
+            mapReduceQueryStatus.setStackTrace(new String(outputStream.toByteArray(), StandardCharsets.UTF_8));
```
Comment (matthpeterson) [inline service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java:184]: Other than the setErrorCode, this appears to be copied code, also used below with the oozie code path  Maybe it's worth adding a function taking an error code

**Review by matthpeterson — APPROVED** [review]

File: service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java (hunk 3 above)
Comment (jwomeara) [reply service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java:184]: i condensed the two exception blocks.

**Review by cogross — APPROVED** [review]

File: service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java (hunk 2 above)
Comment (jwomeara) [reply service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java]: resolved all of the todos.  some got fixed, some don't apply anymore, and some were made into issues.

**Pull request merged**